src/arrayMethodReduce.js

```diff
@@ -4,8 +4,43 @@
  * Implement method Reduce
  */
 function applyCustomReduce() {
-  [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+  [].__proto__.reduce2 = function(callback, ...initialValue) {
+    let initial = initialValue;
+
+    if (initial.length === 0) {
+      initial = undefined;
+    } else {
+      if (initial[0] === undefined) {
+        initial = 'undefined';
+      } else {
+        initial = initial[0];
+      }
+    }
```

Comment:
Also, avoid using nested if statements when it is possible. I'm sure you can do your code without nesting🙂

Reply:
I am sure users can avoid using undefined as values or if they want `'undefined'` they will use `'undefined'` not `undefined`.
But I will change the aproach here.